src/main.rs

```diff
@@ -0,0 +1,1630 @@
+use ::termios::tcgetattr;
+use Flag::*;
+use GeneralPurposeRegister::*;
+use OpCode::*;
+use TrapCode::*;
+use std::{
+    env,
+    fs::File,
+    io::{Read, Write, stdin, stdout},
+    ops::{BitAnd, BitOr, BitOrAssign, Index, IndexMut},
+    os::fd::AsRawFd,
+    path::Path,
+    process::exit,
+};
+use termios::{ECHO, ICANON, TCSANOW, Termios, tcsetattr};
+
+//The number of memory addresses is 2^16
+const MAX_MEMORY_ADDRESS: usize = 1 << 16;
+const MEMORY_REGISTER_KEYBOARD_STATUS_ADDRESS: usize = 0xFE00;
+const MEMORY_REGISTER_KEYBOARD_DATA_ADDRESS: usize = 0xFE02;
+
+const THREE_BIT_MASK: u16 = 0b111;
+const FIVE_BIT_MASK: u16 = 0b11111;
+const SIX_BIT_MASK: u16 = 0b111111;
+const EIGHT_BIT_MASK: u16 = 0xFF;
+const NINE_BIT_MASK: u16 = 0x01FF;
+const TEN_BIT_MASK: u16 = 0x7FF;
```

Comment:
I'd choose one format (binary or hex) and use it in all masks. Also the trailing `0` in the `NINE_BIT_MASK` is redundant